__tests__/_helper.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * This file is part of Serlo.org API
+ *
+ * Copyright (c) 2020 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2020 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/api.serlo.org for the canonical source repository
+ */
+import { RESTDataSource } from 'apollo-datasource-rest'
+import { KeyValueCache } from 'apollo-server-caching'
+import { either } from 'fp-ts'
+import fetch, { RequestInfo, Response, Request } from 'node-fetch'
+
+import { ErrorEvent } from '../src/utils'
+
+export function expectToBeLeftEventWith<A>(
+  value: either.Either<ErrorEvent, A>,
+  expectedEvent: ErrorEvent
+) {
+  expect(either.isLeft(value)).toBe(true)
+
+  if (either.isLeft(value))
+    expect(value.left).toEqual(expect.objectContaining(expectedEvent))
+}
+
+type ResponseSpec = string | Response
+type FetchSpec = Record<string, ResponseSpec>
+
+export function createFetchMock(
+  specs: FetchSpec = {}
+): typeof fetch & FetchMock {
+  const mock = new Proxy(new FetchMock(specs), {
+    apply: (target, _, args: [RequestInfo]) => target.fetch(args[0]),
+  })
+
+  return mock as typeof fetch & FetchMock
+}
+
+export class FetchMock extends Function {
+  private madeRequests: RequestInfo[] = []
+
+  constructor(private specs: FetchSpec) {
+    super()
+  }
```

Comment:
Constructor body can be empty if you don't do anything except super()
```suggestion
  constructor(private specs: FetchSpec) {}
```

Reply:
Then I get the error `Constructors for derived classes must contain a 'super' call.`

Reply:
Ah, okay. Might be that I never had that case when having a base class. Then nvm.